### PR TITLE
feat: add patch to JupyterHub Dockerfile for customization

### PR DESCRIPTION
**Pull request by noeleont**
Allow for customization of JupyterHub, e.g., installing jupyterhub-idle-culler. This patch enables easier extension of JupyterHub functionality.

### CODE REVIEW

File: tutorjupyter/templates/jupyter/build/hub/Dockerfile [hunk 1]
```diff
@@ -12,4 +12,6 @@ RUN pip install \
     jupyterhub-ltiauthenticator==1.6.1 \
     pymysql==1.1.0
 
+{{ patch("jupyter-hub-dockerfile") }}
```
Comment (regisb) [inline tutorjupyter/templates/jupyter/build/hub/Dockerfile:15]: Let's name this "jupyterhub-dockerfile" (see other patches). At the very least, this patch must be documented in the README. Also, please introduce a changelog entry.

Comment (noeleont) [reply tutorjupyter/templates/jupyter/build/hub/Dockerfile:15]: sure, I was orienting myself on `jupyter-lab-dockerfile`. I'll update the PR shortly

Comment (regisb) [reply tutorjupyter/templates/jupyter/build/hub/Dockerfile:15]: :thinking: maybe you're right. I'll let the maintainer decide of that.

Comment (Abdul-Muqadim-Arbisoft) [reply tutorjupyter/templates/jupyter/build/hub/Dockerfile:15]: For consistency with existing patch naming conventions across Tutor plugins, I recommend we rename this patch to "jupyterhub-dockerfile". Additionally, I plan to create a separate PR to update the "jupyter-lab-dockerfile" to "jupyterlab-dockerfile" to maintain uniformity and clarity throughout our plugin architecture.

File: tutorjupyter/__about__.py [hunk 2]
```diff
@@ -1 +1 @@
-__version__ = "18.1.0"
+__version__ = "18.2.0"
```
Comment (DawoudSheraz) [inline tutorjupyter/__about__.py:1]: the version should not be updated here. The maintainer will decide when to release a new version. Please remove this and only add the changelog entry.

**Comment by Abdul-Muqadim-Arbisoft** [comment]
Closing in favor of [PR](https://github.com/overhangio/tutor-jupyter/pull/20)

**Pull request closed without merging**